Update to the latest CTS; run it on all platforms and on all PRs

andyleiserson opened the pull request:
Running it on all PRs was agreed in the wgpu maintainer's meeting.

I've added the directory containing the CTS repo to the GitHub actions cache, since it is big.

Fix a trivial bug in deno to make the CTS work on Linux.

**Testing**
If the CTS job runs and passes on this PR, then it's working.

**Squash or Rebase?** Squash

**Checklist**

- [ ] Run `cargo fmt`.
- [ ] Run `taplo format`.
- [ ] Run `cargo clippy --tests`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
- [ ] Run `cargo xtask test` to run tests.
- [ ] If this contains user-facing changes, add a `CHANGELOG.md` entry. <!-- See instructions at the top of `CHANGELOG.md`. -->

andyleiserson committed:
[deno] Fix typo in feature mapping

Fixes #7751, CTS on Linux

andyleiserson committed:
Update to latest CTS; run it on Windows and Mac and on all PRs

It is not working on Linux.

cwfitzgerald commented on .github/workflows/cts.yml:
```diff
@@ -77,12 +80,10 @@ jobs:
-              echo "=== Running $test ===";
-              DENO_WEBGPU_BACKEND=$backend cargo run --manifest-path ../wgpu/cts_runner/Cargo.toml --frozen -- ./tools/run_deno --verbose "$test";
-            done
+          echo "======= CTS TESTS ======";
+          grep -v '^//' ../wgpu/cts_runner/test.lst | while IFS=$' \t\r\n' read test; do
+            echo "=== Running $test ===";
+            cargo run --manifest-path ../wgpu/cts_runner/Cargo.toml --frozen -- ./tools/run_deno --verbose "$test";
```
Non-blocking thought - we should switch this to use `cargo xtask ctx` for consistency

andyleiserson replied:
The only reason I didn't do this when I added the xtask is because this command line is additionally passing `--frozen` to cargo.

cwfitzgerald replied:
I wouldn't worry about `--frozen` we don't use it anywhere else.

cwfitzgerald commented on .github/workflows/cts.yml:
```diff
@@ -26,13 +23,19 @@ jobs:
           - name: Windows x86_64
             os: windows-2022
             target: x86_64-pc-windows-msvc
-            backends: dx12
```
This can be a follow up: we should make sure DXC is set up

andyleiserson replied:
This might be a third reason to merge the jobs, if `ci.yml` is the model of the necessary DXC setup. Without DXC, is it falling back on FXC?

cwfitzgerald replied:
I'm not sure, it may be falling back on static-dxc - I'd need to look at the code to know for specific.

cwfitzgerald reviewed (approved):
This is so cooool!

cwfitzgerald commented:
Filed https://github.com/gfx-rs/wgpu/issues/7753 for code coverage as well

andyleiserson committed:
Use `cargo xtask cts`

cwfitzgerald commented:
The CI issue is a bug in the "disable debug" code, as it overwrites the whole file instead of appending that didn't matter until now

andyleiserson commented:
Thank you. Is the triple quote doing something?

andyleiserson committed:
Fix redirection

andyleiserson committed:
Tweak caching, fix path handling on Windows, and fetch the CTS if needed

andyleiserson commented:
My first attempt at using `cargo xtask cts` was bunk. I believe it's good now, I also made a couple other changes:

* Add the CTS checkout to the cache.
* Add a git fetch to the xtask in the case where a clone is already present but doesn't have the desired revision.
* Fixed a path handling problem on Windows. `canonicalize` produces a "verbatim path", which cargo does not like, so I switched to `std::path::absolute` instead.

cwfitzgerald reviewed (approved):
Looks good! Lets land and iterate!
